Sources/StreamChatUI/Components.swift

```diff
@@ -302,6 +302,22 @@ public struct Components {
 
     /// The view that shows a number of unread messages in channel.
     public var channelUnreadCountView: ChatChannelUnreadCountView.Type = ChatChannelUnreadCountView.self
+    
+    /// The view that is displayed when there are no channels on the list, i.e. when is on empty state.
+    public var channelListEmptyView: ChatChannelListEmptyView.Type = ChatChannelListEmptyView.self
+    
+    /// View that shows that some error occurred on ChatChannelList.
+    public var channelListErrorView: ChatChannelListErrorView.Type = ChatChannelListErrorView.self
+    
+    /// The `UICollectionViewCell` subclass that shows while loading the channel list.guard let self = self else { return }
+    public var channelListCollectionViewSkeletonCell: ChatChannelListCollectionViewSkeletonCell.Type = ChatChannelListCollectionViewSkeletonCell.self
+    
+    /// View that shows the loading state of a channel list item
+    public var channelListSkeletonItemView: ChatChannelListItemSkeletonView.Type = ChatChannelListItemSkeletonView.self
+    
+    /// A boolean value that determines whether the loading state on Channel list is shown as a Skeleton view.
+    /// True by default. If set to false, default loading indicator is shown instead.
+    public var isSkeletonViewEnabled = true
```

Comment:
Didn't we say this should be false by default? Only in the demo app we will set it to true.

Reply:
@hugobernalstream, the name is also not correct. Ideally we have a bool for enabling/disabling the channel states altogether. As mentioned before, customers right now might have their own state handling already, with error views etc... So Ideally this should be something like `isChannelListStatesEnabled: Bool` or have 2 flags like `isChannelListLoadingStateEnabled: Bool` and `isChannelListErrorStateEnabled: Bool`. 

We can't name it "Skeleton" because in the future we can even change this. The skeleton view is an implementation detail.

And like @martinmitrevski said, this should be disabled by default.

Reply:
Done 👍🏻